examples/getting_started/quickstart.ipynb

```diff
@@ -0,0 +1,535 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Mirascope Quickstart Guide\n",
+    "\n",
+    "1. [Setup](#Setup)\n",
+    "2. [Prompt Templates](#Prompt-Templates)\n",
+    "3. [Basic LLM Call](#Basic-LLM-Call)\n",
+    "4. [Streaming Responses](#Streaming-Responses)\n",
+    "5. [Asynchronous Processing](#Asynchronous-Processing)\n",
+    "6. [JSON Mode](#JSON-Mode)\n",
+    "7. [Response Models](#Response-Models)\n",
+    "8. [Output Parsers](#Output-Parsers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "Mirascope supports various LLM providers, including [OpenAI](https://openai.com/), [Anthropic](https://www.anthropic.com/), [Mistral](https://mistral.ai/), [Gemini](https://gemini.google.com), [Groq](https://groq.com/), [Cohere](https://cohere.com/), [LiteLLM](https://www.litellm.ai/), [Azure AI](https://azure.microsoft.com/en-us/solutions/ai), and [Vertex AI](https://cloud.google.com/vertex-ai). For the purposes of this guide, we will be using OpenAI. Let's start by installing Mirascope and its dependencies:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install \"mirascope[openai]\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This command installs Mirascope along with the necessary packages for the OpenAI integration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"YOUR_API_KEY\"\n",
+    "# Set the appropriate API key for the provider you're using"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prompt Templates\n",
+    "\n",
+    "Prompt templates in Mirascope allow you to create dynamic and reusable prompts."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2024-09-06T14:00:06.593677Z",
+     "start_time": "2024-09-06T14:00:06.590420Z"
+    }
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "None\n"
+     ]
+    }
+   ],
+   "source": [
+    "from mirascope.core import prompt_template\n",
+    "\n",
+    "\n",
+    "@prompt_template(\"What is the capital of {country}?\")\n",
+    "def get_capital_prompt(country: str): ...\n",
+    "\n",
+    "\n",
+    "print(get_capital_prompt(\"Japan\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example:\n",
+    "1. The `@prompt_template` decorator defines the structure of the prompt.\n",
+    "2. `{country}` is a placeholder that gets replaced with the matching argument passed to the function.\n",
+    "\n",
+    "Here's an example of a multi-line prompt:"
```

Comment:
this is not multi-line, this is using messages with different roles using the role keywords

multiline prompt is when the content for a single message spans multiple lines (and needs to be started on a newline in order to be parsed properly).

For example:

## BAD

These will be parsed incorrectly and include erroneous tabs since cleandoc/dedent will have no indentation from the first line of the multi-line prompt

```python
# BAD, will be parsed incorrectly and include erroneous tabs

@prompt_template(
    """First line
    Second line
    """
)

@prompt_template(
    """
    USER: First line
    Second line
    """
)

@prompt_template(
    """
    SYSTEM: First line
    Second line

    USER: single line prompt
)
```

## GOOD

```python
@prompt_template(
    """
    First line
    Second line
    """
)

@prompt_template(
    """
    USER:
    First line
    Second line
    """
)

@prompt_template(
    """
    SYSTEM:
    First line
    Second line

    USER: single line prompt
)
```